azurerm_app_service/vars.tf

```
@@ -0,0 +1,101 @@
+variable "global_prefix" {
+  type = string
+}
+
+variable "environment" {
+  type = string
+}
+
+variable "environment_short" {
+  type = string
+}
+
+variable "region" {
+  type = string
+}
+
+variable "name" {
+  type = string
+}
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "log_analytics_workspace_id" {
+  type        = string
+  description = "The log_analytics workspace id"
+}
+
+variable "storage_account_info" {
+  type = object({
+    account_tier             = string
+    account_replication_type = string
+    access_tier              = string
+  })
+
+  default = {
+    account_tier             = "Standard"
+    account_replication_type = "LRS"
+    access_tier              = "Hot"
+  }
+}
+
+variable "app_service_plan_info" {
+  type = object({
+    kind     = string
+    sku_tier = string
+    sku_size = string
+  })
+
+  default = {
+    kind     = "elastic"
+    sku_tier = "ElasticPremium"
+    sku_size = "EP1"
+  }
+}
+
+variable virtual_network_info {
```

most likely they are working but according to this documentation:
https://learn.hashicorp.com/terraform/azure/variables_az#defining-variables variable name should be in double quote.
```suggestion
variable "virtual_network_info" {
```